django/thunderstore/community/models/package_listing.py

```diff
@@ -101,6 +103,13 @@ def validate(self):
 
     def save(self, *args, **kwargs):
         self.validate()
+
+        old_self = PackageListing.objects.filter(pk=self.pk).first()
+        if old_self is None:
+            self.update_visibility()
```

Comment:
This should be possible to replace with `if self.pk is None:` rather than querying the DB, so if you didn't test that yet I'd suggest doing so (unsaved models don't have a `pk` set in Django)